<launch>
    <group ns="serial">
        <node name="rosserial" pkg="rosserial_python" type="serial_node.py">
		    <param name="port" value="/dev/ttyAMA0"/>
		    <param name="baud" value="333333.33"/>
	    </node>
    </group>

	<!-- <node pkg="tf" type="static_transform_publisher" name="imu_to_base_link" args="0.063 -0.015 -0.036 0 0 0 imu_data base_link 100" /> -->
	<node pkg="tf" type="static_transform_publisher" name="odom_to_base_link" args="0.0 0.0 0.0 0 0 0 odom base_link 100" />
	<node pkg="tf" type="static_transform_publisher" name="to_uss_front" args="0.076 0.035 0.09 0 0 0 base_link uss_data 100" />

	<!-- <node pkg="run_ruediger" type="imu_tf2_listener" name="listener" /> -->
	<node 
		pkg		=	"run_ruediger" 
		type	=	"tf2_manager" 
		name	=	"tf2_manager" 
	>
		<param name="imu_topic" value="/serial/imu_data"/>
		<param name="imu_frame_id" value="/world"/>
		<param name="imu_child_frame_id" value="/imu_data"/>

		<param name="odom_mot_left_topic" value="/serial/encoder_2"/>
		<param name="odom_mot_right_topic" value="/serial/encoder_3"/>
		<param name="odom_frame_id" value="/world"/>
		<param name="odom_child_frame_id" value="/odom"/>

		<param name="init_x" value="1.30" />
		<param name="init_y" value="0.75" />
		<param name="init_z" value="0.0" />
		<param name="init_roll" value="0.0" />
		<param name="init_pitch" value="0.0" />
		<param name="init_yaw" value="-1.570796327" />
		<!-- <param name="init_yaw" value="-M_PI/2" /> -->
	</node>

</launch>
